src/Microsoft.AspNetCore.Http/Internal/DefaultHttpRequest.cs

```diff
@@ -41,7 +41,7 @@ public virtual void Uninitialize()
             _features.Fetch(ref _features.Cache.Query, f => new QueryFeature(f));
 
         private IFormFeature FormFeature =>
-            _features.Fetch(ref _features.Cache.Form, this, f => new FormFeature(f));
+            _features.Fetch(ref _features.Cache.Form, this, f => new FormFeature(f, (f._context as DefaultHttpContext)?.FormOptions));
```

Comment:
@henkmollema Updated with state capturing; does increase the size by 16 bytes as both `DefaultHttpReXX` and `DefaultHttpContext` need the reference due to the way initialisation works